docs_src/design/adr/security/tbd-authentication-e2ee.md

```diff
@@ -0,0 +1,414 @@
+# EdgeX Microservice Authentication (end-to-end security)
+
+### Submitters
+
+- Bryon Nevis (Intel)
+- Clint Dovholuk (NetFoundry - OpenZiti)
+
+## Change Log
+
+- [proposed](https://github.com/edgexfoundry/edgex-docs/pull/935) (2023-01-28)
+
+## Referenced Use Case(s)
+
+- [Microservice Authentication](https://docs.edgexfoundry.org/3.0/design/ucr/Microservice-Authentication/)
+
+
+## Context
+
+The AS-IS Architecture figure below depicts the current state of
+microservice communication security prior to EdgeX 3.0,
+when security is enabled:
+
+![AS-IS Architecture](tbd-as-is.jpg)
+
+As shown in the diagram,
+many of the foundational services used by EdgeX Foundry
+have already been secured:
+
+* Communication with EdgeX's secret store, as implemented by
+  Hashicorp Vault, is secured over a local HTTP socket with
+  token-based authentication.  An access control list limits access
+  to the keyspace of the key value store.
+
+* Communication with EdgeX's service registry and configuration provider,
+  as implemented by Hashicorp Consul, is secured over a local HTTP
+  socket with token-based authentication,
+  with the token being mediated by Hashicorp Vault.
+  An access control list limits access to the keyspace of
+  the configuration store.
+
+* Communication with EdgeX's default database, Redis, is secured using
+  username/password authentication, with the password stored
+  in Hashicorp Vault.  An access control list limits the commands
+  that clients are allowed to issue to the server.
+
+External access to EdgeX microservices has also been secured.
+EdgeX microservices only bind to local ports,
+and are only exposed externally through a Kong API gateway.
```

Comment:
This paragraphs that reference Kong need to be rewrite for NGINX

Reply:
Rewritten for current current state.